jscomp/stdlib-402/pervasives.ml

```diff
@@ -268,10 +268,14 @@ external snd : 'a * 'b -> 'b = "%field1"
 
 type 'a ref = { mutable contents : 'a }
 external ref : 'a -> 'a ref = "%makemutable"
-external ( ! ) : 'a ref -> 'a = "%field0"
-external ( := ) : 'a ref -> 'a -> unit = "%setfield0"
-external incr : int ref -> unit = "%incr"
-external decr : int ref -> unit = "%decr"
+(* external ( ! ) : 'a ref -> 'a = "%field0" *)
+let (!) x = x.contents
```

Comment:
maybe we can introduce new primitives for deferencing and update

Reply:
This may affect an optimization pass of reference elimination

Reply:
Sure, can be done in a cleanup phase. Reminder: search `field0` and create another version.
In this pass, I'd like to focus on correctness first.